src/types.rs

```diff
@@ -47,6 +47,13 @@ pub enum Status {
     InternalFailure = 10,
 }
 
+#[repr(u32)]
+#[derive(Debug)]
+pub enum ContextType {
+    HttpContext = 0,
```

Comment:
Could you re-add `RootContext` as `0` here? Thanks.

Reply:
Why? How would that be different from returning `None`?